lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1963,5 +1963,61 @@ sub add_to_temporary_input_files_queue {
     return 1;
 }
 
+sub lock_bam_file_access {
+    my $self = shift;
+
+    unless ($self->get_merged_alignment_results) {
+        my @bams = $self->alignment_bam_file_paths;
+        unless (@bams) {
+            die $self->error_message("Alignment result with class (%s) and id (%s) has neither ".
```

Comment:
That's even worse!  This process gets the bam file path here, and then the merged alignment step commits and releases the lock, then we get the lock.  Now we have a path to a bam file that will be deleted out from under us.